Allow to suppress charging state notification

wdehoog opened the pull request:
Some people don't like charging state indication and some devices might only have one led blink pattern. These changes allow to disable showing them (dconf write ```/com/lomiri/ledindication/charging-state-visible false```).

See https://github.com/ubports/ubuntu-touch/issues/1377

(edited to show new settings namespace)

wdehoog committed:
allow to suppress charging state notification

wdehoog committed:
fix whitespace

wdehoog committed:
added setting for IndicatorsLight test

wdehoog committed:
added test cases for charge-state-visibility setting

wdehoog committed:
update when settings has changed

wdehoog committed:
added missing signal

UniversalSuperBox commented:
Is your plan to add these settings somewhere in `system-settings` as well?

Since some devices only have a single-color LED, I'd recommend adding something to the device config parser in another PR. This additional metadata could tell the system if it should even try to use multi-color LEDs. Then users don't need to choose if they want their LED on all the time during charging.

mariogrip commented on data/com.canonical.Unity8.gschema.xml:
```diff
@@ -72,4 +72,12 @@
       <description>If too many unsuccessful login attempts have been made in a row, the greeter locks you out for a while.  This unix timestamp indicates when you were locked out, so unity8 knows when to let you back in.</description>
     </key>
   </schema>
+
```
I suggest when adding new gsettings use the com.lomiri userspace to ease with transition later on.

UniversalSuperBox replied:
Is that possible when we're in the com.canonical.Unity.gschema.xml file?

UniversalSuperBox replied:
Alright, now there's a new com.lomiri.gschema.xml file. @wdehoog, is the change in this file needed any more?

wdehoog committed:
use com.lomiri namespace for new setting

wdehoog committed:
remove unused schema

wdehoog committed:
use createTemporaryQmlObject instead of a Loader to create the object to test.
The combination Loader/Component results in another instance in memory which also starts to
control the led making testing impossible.

wdehoog committed:
whitespace and a ;

wdehoog committed:
two new test cases

UniversalSuperBox commented:
Sorry for the wait... Unfortunately, this PR now conflicts on fake_gsettings... Could you please resolve this conflict (you may want to `git pull --rebase upstream xenial`) so we can take another look?
